src/main/java/bowling/domain/State.java

```diff
@@ -0,0 +1,100 @@
+package bowling.domain;
+
+import java.util.Arrays;
+
+public enum State {
+    GUTTER("-", 0, false),
+    READY("0",0,false),
+    ONE("1", 1, false),
+    TWO("2", 2, false),
+    THREE("3", 3, false),
+    FOUR("4", 4, false),
+    FIVE("5", 5, false),
+    SIX("6", 6, false),
+    SEVEN("7", 7, false),
+    EIGHT("8", 8, false),
+    NINE("9", 9, false),
+    STRIKE("X", 10, false),
+    SPARE("/", 10, true);
+
+    private final String value;
+    private final int fallenPins;
+    private final boolean isSpare;
+
+    State(String value, int fallenPins, boolean isSpare) {
+        this.value = value;
+        this.fallenPins = fallenPins;
+        this.isSpare = isSpare;
+    }
+
+    public static State valueOf(int fallenPins, boolean isSpare) {
+        return Arrays.stream(State.values())
+                .filter(value -> value.fallenPins == fallenPins && value.isSpare == isSpare)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("올바른 값이 아닙니다."));
+    }
+
+    public static State bowl(int previousFallenPins, int nextFallenPins, int statesLength) {
+        if (checkNormalFrameStrike(nextFallenPins, statesLength)) return STRIKE;
+
+        if (checkGutter(nextFallenPins)) return GUTTER;
+
+        validateFallenPinSum(previousFallenPins, nextFallenPins);
+
+        if (checkNormalFrameSpare(previousFallenPins, nextFallenPins)) return SPARE;
```

Comment:
조건 수행문이 한줄이라도 괄호를 붙여주시는게 컨벤션이오니 참고 부탁드립니다 :)
```suggestion
        if (checkNormalFrameStrike(nextFallenPins, statesLength)) {
            return STRIKE;
        }

        if (checkGutter(nextFallenPins)) {
            return GUTTER;
        }   

        validateFallenPinSum(previousFallenPins, nextFallenPins);

        if (checkNormalFrameSpare(previousFallenPins, nextFallenPins)) {
            return SPARE;
        }
```